src/main/java/edu/cornell/kfs/module/purap/document/CuPaymentRequestDocument.java

```diff
@@ -272,6 +275,19 @@ public void doActionTaken(ActionTakenEvent event) {
             }
         }
 
+    @Override
+    public List<String> getWorkflowEngineDocumentIdsToLock() {
+        Stream<String> otherDocIdsToLock = Stream.of(super.getWorkflowEngineDocumentIdsToLock())
+                .flatMap(idList -> (idList != null) ? idList.stream() : Stream.empty());
+        

```

Comment:
Base code has special processing in place for split PO's in class org.kuali.kfs.module.purap.document.PurchaseOrderDocument line 605. Do we need to be concerned about that here?

That class also has code in place on line 616 to allow the output of the documentNumber when debug processing is turned on. Should we consider having that in place in this method as well?

Reply:
I don't anticipate any changes being needed when a PO Split Document is involved. If possible, though, it would be good to have such a situation tested to make sure the PREQs handle it accordingly, and then we can make future adjustments if issues are discovered there.

As for the document number logging, the StandardWorkflowEngine.lockAdditionalDocuments() method already logs the document IDs of the docs that it attempts to lock. However, if we decide to update the fix to lock more than one document, then we could look into adding some debug logging to show all the relevant doc IDs.